dll/win32/shell32/CShellLink.cpp

```diff
@@ -2891,13 +2924,21 @@ INT_PTR CALLBACK CShellLink::SH_ShellLinkDlgProc(HWND hwndDlg, UINT uMsg, WPARAM
 
                     if (pThis->m_sIcoPath)
                         wcscpy(wszPath, pThis->m_sIcoPath);
+                    else
+                        FindExecutableW(pThis->m_sPath, NULL, wszPath);
+
                     INT IconIndex = pThis->m_Header.nIconIndex;
                     if (PickIconDlg(hwndDlg, wszPath, _countof(wszPath), &IconIndex))
                     {
                         pThis->SetIconLocation(wszPath, IconIndex);
-                        ///
-                        /// FIXME redraw icon
-                        ///
+
+                        HICON hIconLarge = NULL;
+                        if (S_OK == pThis->Extract(wszPath, IconIndex, &hIconLarge, NULL, 1))
+                        {
+                            HICON hIconOld = (HICON)SendDlgItemMessageW(hwndDlg, 14000, STM_GETICON, 0, 0);
+                            SendDlgItemMessageW(hwndDlg, 14000, STM_SETICON, (WPARAM)hIconLarge, 0);
```

Comment:
Maybe using a resource name instead of "14000" would be better :D

Reply:
Can I use <dlgs.h> resource IDs instead of user defined IDC_*?

Reply:
OK.